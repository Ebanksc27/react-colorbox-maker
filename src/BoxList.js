import React, { useState } from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';

function BoxList() {
  // Initial state will be an empty array for boxes
  const [boxes, setBoxes] = useState([]);

  const addBox = newBox => {
    setBoxes(oldBoxes => [...oldBoxes, newBox]);
  };

  const removeBox = id => {
    setBoxes(oldBoxes => oldBoxes.filter(box => box.id !== id));
  };

  // Render the Box components and NewBoxForm
  return (
    <div>
      <NewBoxForm addBox={addBox} />
      {boxes.map(box => (
        <Box key={box.id} {...box} removeBox={removeBox} />
      ))}
    </div>
  );
}

export default BoxList;
