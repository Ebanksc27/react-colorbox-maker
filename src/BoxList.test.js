import React from 'react';
import { render } from '@testing-library/react';
import BoxList from './BoxList';

test('renders BoxList component without crashing', () => {
  render(<BoxList />);
});

test('BoxList component snapshot', () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});
