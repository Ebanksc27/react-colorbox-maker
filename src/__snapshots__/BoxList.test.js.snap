// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxList component snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        name="width"
        placeholder="Width"
        type="text"
        value=""
      />
      <input
        name="height"
        placeholder="Height"
        type="text"
        value=""
      />
      <input
        name="backgroundColor"
        placeholder="Background Color"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add a new box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
